service/controller/azuremachinepool/handler/migration/azuremachinepool_test.go

```diff
@@ -0,0 +1,308 @@
+package migration
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	oldcapzexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capzexp/v1alpha3"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1"
+
+	"github.com/giantswarm/azure-operator/v5/service/unittest"
+)
+
+func TestEnsureCreatedAzureMachinePoolIsCorrect(t *testing.T) {
```

Comment:
Test that checks if new AzureMachinePool CR is correct. It uses golden files for expected CRs data.